File: dart/unit_test/demo/test/calculator_test.dart
import 'package:test/test.dart';
import '../lib/calculator.dart';

void main() {
  group('Calculator', () {
    final calculator = Calculator();

    test('adds two numbers', () {
      print('Testing add...');
      expect(calculator.add(2, 3), equals(5));
    });

    test('subtracts two numbers', () {
      print('Testing subtract...');
      expect(calculator.subtract(5, 3), equals(2));
    });
  });
}

File: kotlin/unit_test/demo/src/test/kotlin/CalculatorTest.kt
import kotlin.test.Test
import kotlin.test.assertEquals

class CalculatorTest {

    private val calculator = Calculator()

    @Test
    fun testAdd() {
        assertEquals(5, calculator.add(2, 3))
    }

    @Test
    fun testSubtract() {
        assertEquals(3, calculator.subtract(5, 2))
    }
}